src/python/pants/option/global_options.py

```diff
@@ -1511,6 +1511,8 @@ def compute_pantsd_invalidation_globs(
                 # macros should be adapted to allow this dependency to be automatically detected.
                 "requirements.txt",
                 "3rdparty/**/requirements.txt",
+                "pyproject.toml",
+                "3rdparty/**/pyproject.toml",
```

Comment:
I doubt anyone will put a `pyproject.toml` under `3rdparty`, but it doesn't hurt to have this.